### PR TITLE
Display RTC on OSD

### PR DESCRIPTION
**Pull request by RisingFog**
This adds an option in the Movie menu to show the current system clock.

Meant to compliment the recent Custom RTC commits being merged.

---
This change is [Reviewable](https://reviewable.io/reviews/dolphin-emu/dolphin/4036)

### CODE REVIEW

File: Source/Core/Core/Movie.cpp [hunk 1]
```diff
@@ -179,6 +179,18 @@ std::string GetInputDisplay()
 }
 
 // NOTE: GPU Thread
+std::string GetRTCDisplay()
+{
+  time_t currentTime = CEXIIPL::GetGCTime() + 0x386D4380;  // Seconds between 1.1.1970 and 1.1.2000
+  tm* gmTime = gmtime(&currentTime);
+  char buffer[256];
+  strftime(buffer, sizeof(buffer), "Date/Time: %c", gmTime);
```
Comment (lioncash) [inline Source/Core/Core/Movie.cpp]: [std::put_time](http://en.cppreference.com/w/cpp/io/manip/put_time)

File: Source/Core/Core/Movie.cpp [hunk 2]
```diff
@@ -179,6 +179,18 @@ std::string GetInputDisplay()
 }
 
 // NOTE: GPU Thread
+std::string GetRTCDisplay()
+{
+  time_t currentTime = CEXIIPL::GetGCTime() + 0x386D4380;  // Seconds between 1.1.1970 and 1.1.2000
```
Comment (lioncash) [inline Source/Core/Core/Movie.cpp]: `current_time` and `gm_time`

**Comment by RisingFog** [comment]
Review status: 0 of 9 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[Source/Core/Core/Movie.cpp, line 184 \[r1\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/4036#-KN4g8JosC5DgR-o4lmS-r1-184:-KN56PCOcVQe5P2iAMoi:-499146009) ([raw file](https://github.com/dolphin-emu/dolphin/blob/718d8e570cc25e87ebcd5b30ad630cf41dbe3266/Source/Core/Core/Movie.cpp#L184)):*
<details><summary><i>Previously, lioncash (Mat M.) wrote…</i></summary>
> `current_time` and `gm_time`
</details>
Done.

---

*[Source/Core/Core/Movie.cpp, line 187 \[r1\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/4036#-KN4g8JosC5DgR-o4lmS-r1-187:-KN56PZ7UtiSnhX7152L:-499146009) ([raw file](https://github.com/dolphin-emu/dolphin/blob/718d8e570cc25e87ebcd5b30ad630cf41dbe3266/Source/Core/Core/Movie.cpp#L187)):*
<details><summary><i>Previously, lioncash (Mat M.) wrote…</i></summary>
> [std::put_time](http://en.cppreference.com/w/cpp/io/manip/put_time)
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/dolphin-emu/dolphin/4036)*

**Comment by JosJuice** [comment]
Review status: 0 of 9 files reviewed at latest revision, 4 unresolved discussions.

---

*[Source/Core/Core/Movie.cpp, line 184 \[r2\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/4036#-KN6x4VrGntLoy9bNI32:-KN6x4VsEbDRJp9C_Mkt:1472444346) ([raw file](https://github.com/dolphin-emu/dolphin/blob/5bc1d2ecdec40aff99e51a4aefeb43eac0a9f6b0/Source/Core/Core/Movie.cpp#L184)):*
> ```cpp
> std::string GetRTCDisplay()
> {
>   time_t current_time = CEXIIPL::GetGCTime() + 0x386D4380;  // Seconds between 1.1.1970 and 1.1.2000
> ```

If you're going to use `0x386D4380`, please make it a named constant. You could put it in EXI_DeviceIPL and use it both here and in `CEXIIPL::GetGCTime`.

---

*[Source/Core/Core/Movie.h, line 182 \[r2\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/4036#-KN6xbsQKewSEmz_EvLw:-KN6xbsQKewSEmz_EvLx:-757154079) ([raw file](https://github.com/dolphin-emu/dolphin/blob/5bc1d2ecdec40aff99e51a4aefeb43eac0a9f6b0/Source/Core/Core/Movie.h#L182)):*
> ```objc
> 
> std::string GetInputDisplay();
> std::string GetRTCDisplay();
> ```

Why is this in Movie?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/dolphin-emu/dolphin/4036)*

**Comment by JosJuice** [comment]
Review status: 0 of 9 files reviewed at latest revision, 4 unresolved discussions.

---

*[Source/Core/Core/Movie.cpp, line 184 \[r2\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/4036#-KN6x4VrGntLoy9bNI32:-KN6y5jckX4r27pPtVV4:138555854) ([raw file](https://github.com/dolphin-emu/dolphin/blob/5bc1d2ecdec40aff99e51a4aefeb43eac0a9f6b0/Source/Core/Core/Movie.cpp#L184)):*
<details><summary><i>Previously, JosJuice wrote…</i></summary>
> If you're going to use `0x386D4380`, please make it a named constant. You could put it in EXI_DeviceIPL and use it both here and in `CEXIIPL::GetGCTime`.
</details>
Or you could split `GetGCTime` into a `GetTime` function that gives you want you want here and a `GetGCTime` function that just runs `return GetTime() - 0x386D4380` (preferably still with a named constant, but the constant can be function-local this way)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/dolphin-emu/dolphin/4036)*

**Comment by RisingFog** [comment]
Review status: 0 of 11 files reviewed at latest revision, 4 unresolved discussions.

---

*[Source/Core/Core/Movie.cpp, line 187 \[r1\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/4036#-KN4g8JosC5DgR-o4lmS-r1-187:-KNEFOskmDe5oD4Bz90n:-358160392) ([raw file](https://github.com/dolphin-emu/dolphin/blob/718d8e570cc25e87ebcd5b30ad630cf41dbe3266/Source/Core/Core/Movie.cpp#L187)):*
<details><summary><i>Previously, RisingFog (Chris Burgener) wrote…</i></summary>
> Done.
</details>
Reverted because std::put_time does not exist in GCC 4.9.

---

*[Source/Core/Core/Movie.cpp, line 184 \[r2\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/4036#-KN6x4VrGntLoy9bNI32:-KNEEgYk2cRPDo5RwK68:-991891367) ([raw file](https://github.com/dolphin-emu/dolphin/blob/5bc1d2ecdec40aff99e51a4aefeb43eac0a9f6b0/Source/Core/Core/Movie.cpp#L184)):*
<details><summary><i>Previously, JosJuice wrote…</i></summary>
> Or you could split `GetGCTime` into a `GetTime` function that gives you want you want here and a `GetGCTime` function that just runs `return GetTime() - 0x386D4380` (preferably still with a named constant, but the constant can be function-local this way)
</details>
Created a GetCTTimeJan1970() function which returns GetGCTime() + cJanuary2000 (the constant for seconds between Jan 1st 1970 and Jan 1st 2000).

---

*[Source/Core/Core/Movie.h, line 182 \[r2\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/4036#-KN6xbsQKewSEmz_EvLw:-KNEEtFy-bP3TaMwyqae:733565661) ([raw file](https://github.com/dolphin-emu/dolphin/blob/5bc1d2ecdec40aff99e51a4aefeb43eac0a9f6b0/Source/Core/Core/Movie.h#L182)):*
<details><summary><i>Previously, JosJuice wrote…</i></summary>
> Why is this in Movie?
</details>
Because the toggle is kept within the movie function, most calls to the OSD are done within Movie or use Movie functions and the includes needed are already defined.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/dolphin-emu/dolphin/4036)*

**Comment by BhaaLseN** [comment]
Review status: 0 of 11 files reviewed at latest revision, 6 unresolved discussions.

---

*[Source/Core/Core/HW/EXI_DeviceIPL.cpp, line 35 \[r3\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/4036#-KNHoMeycYedGDq5IyRC:-KNHoMezxwgARWOPitl2:-19115422) ([raw file](https://github.com/dolphin-emu/dolphin/blob/db3b593d3bf9c077d0d61d1b0edbb23db027c268/Source/Core/Core/HW/EXI_DeviceIPL.cpp#L35)):*
> ```cpp
>                                       "(C) 1999 ArtX Inc.  All rights reserved.";
> 
> static const u32 cJanuary2000 = 0x386D4380;  // Seconds between 1.1.1970 and 1.1.2000
> ```

Does it make sense to use `constexpr` instead?

---

*[Source/Core/VideoCommon/RenderBase.cpp, line 338 \[r3\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/4036#-KNHofSRgRBxTLixVRVB:-KNHofSSQJR7KSkenh9s:831227915) ([raw file](https://github.com/dolphin-emu/dolphin/blob/db3b593d3bf9c077d0d61d1b0edbb23db027c268/Source/Core/VideoCommon/RenderBase.cpp#L338)):*
> ```cpp
>   if (SConfig::GetInstance().m_ShowRTC)
>   {
>     final_cyan += Movie::GetRTCDisplay();
> ```

Not exactly knowing what it looks like on screen, I'd say this lacks the extra newline for `final_yellow` that all the others have. Could also be that you don't need it here, and this comment can be ignored.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/dolphin-emu/dolphin/4036)*

**Comment by RisingFog** [comment]
Review status: 0 of 11 files reviewed at latest revision, 6 unresolved discussions, some commit checks failed.

---

*[Source/Core/Core/HW/EXI_DeviceIPL.cpp, line 35 \[r3\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/4036#-KNHoMeycYedGDq5IyRC:-KNHtQghIa3PoUhtZ1XX:-499146009) ([raw file](https://github.com/dolphin-emu/dolphin/blob/db3b593d3bf9c077d0d61d1b0edbb23db027c268/Source/Core/Core/HW/EXI_DeviceIPL.cpp#L35)):*
<details><summary><i>Previously, BhaaLseN (BhaaL) wrote…</i></summary>
> Does it make sense to use `constexpr` instead?
</details>
Done.

---

*[Source/Core/VideoCommon/RenderBase.cpp, line 338 \[r3\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/4036#-KNHofSRgRBxTLixVRVB:-KNHtRvo9_WO0bZwn-S2:-499146009) ([raw file](https://github.com/dolphin-emu/dolphin/blob/db3b593d3bf9c077d0d61d1b0edbb23db027c268/Source/Core/VideoCommon/RenderBase.cpp#L338)):*
<details><summary><i>Previously, BhaaLseN (BhaaL) wrote…</i></summary>
> Not exactly knowing what it looks like on screen, I'd say this lacks the extra newline for `final_yellow` that all the others have. Could also be that you don't need it here, and this comment can be ignored.
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/dolphin-emu/dolphin/4036)*

**Comment by Aestek** [comment]
You could use `OSD::AddMessage` api for this which is cleaner IMHO

**Comment by RisingFog** [comment]
@Aestek OSD::AddMessage isn't for persistent OSD messages.

**Comment by RisingFog** [comment]
Is there anything preventing this from review/merge?

**Pull request merged**